app/models/menu/menu.rb

```diff
@@ -0,0 +1,420 @@
+        title: Translation.translate('My Patients'),
+      ),
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { !(user.can_view_patients_for_own_agency? && user.health_agencies.any?) && user.can_view_some_vprs? },
+        path: health_flexible_service_my_vprs_path,
+        title: Translation.translate('My Patients'),
+      ),
+    )
+
+    menu
+  end
+
+  def data_menu
+    menu = Menu::Item.new(
+      user: user,
+      title: Translation.translate('Data'),
+      icon: 'icon-spin1',
+      id: 'data',
+    )
+    path = ad_hoc_data_sources_path
+    path = data_sources_path if (user.can_view_imports_projects_or_organizations? || user.can_upload_hud_zips? || user.can_manage_some_ad_hoc_ds?) && (user.can_view_projects? || user.can_view_organizations? || user.can_upload_hud_zips?)
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_view_imports_projects_or_organizations? || user.can_upload_hud_zips? || user.can_manage_some_ad_hoc_ds? },
+        path: path,
+        title: Translation.translate('Data Sources'),
+      ),
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_see_raw_hmis_data? },
+        path: source_data_path,
+        title: Translation.translate('HMIS Source Data'),
+      ),
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_edit_some_project_groups? },
+        path: project_groups_path,
+        title: Translation.translate('Project Groups'),
+      ),
+    )
+
+    menu
+  end
+
+  def administration_menu
+    menu = Menu::Item.new(
+      user: user,
+      title: Translation.translate('Administration'),
+      icon: 'icon-cog',
+      id: 'administration',
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { user.can_edit_users? || user.can_edit_translations? || user.can_view_imports? },
+        path: user.admin_dashboard_landing_path,
+        title: Translation.translate('Warehouse Admin'),
+      ),
+    )
+    path = if user.can_approve_patient_assignments?
+      review_admin_health_patient_referrals_path
+    elsif (user.can_manage_health_agency? || user.can_manage_patients_for_own_agency?) && user.health_agencies.any?
+      review_admin_health_agency_patient_referrals_path
+    else
+      admin_health_admin_index_path
+    end
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(user) { GrdaWarehouse::Config.get(:healthcare_available) && (user.can_administer_health? || user.can_manage_health_agency? || user.has_patient_referral_review_access?) },
+        path: path,
+        title: Translation.translate('Healthcare Admin'),
+      ),
+    )
+    menu.add_child(
+      Menu::Item.new(
+        user: user,
+        visible: ->(_user) { context.hmis_admin_visible? },
+        path: hmis_admin_users_path,
```

Comment:
When ENABLE_HMIS_API is not set, I am running into the following error wehn trying to load the application: "undefined local variable or method `hmis_admin_users_path'". Setting ENABLE_HMIS_API and HMIS_HOSTNAME does resolve the issue.

Reply:
Thank you!  I'll look to see if that's fixable.